packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -295,17 +291,14 @@ export class IndexedDbQueryCache implements QueryCache {
   ): PersistencePromise<void> {
     // PORTING NOTE: The reverse index (documentsTargets) is maintained by
     // IndexedDb.
-    const promises: Array<PersistencePromise<void>> = [];
     const store = documentTargetStore(txn);
-    keys.forEach(key => {
+    return PersistencePromise.waitForEach(keys, key => {
       const path = EncodedResourcePath.encode(key.path);
-      promises.push(store.delete([targetId, path]));
-      if (this.garbageCollector !== null) {
-        this.garbageCollector.addPotentialGarbageKey(key);
-      }
-      promises.push(this.referenceDelegate.removeReference(txn, key));
+      return PersistencePromise.waitFor([
+        store.delete([targetId, path]),
+        this.referenceDelegate.removeReference(txn, key)
+      ]);
```

Comment:
This could just be:
```
return store.delete(..).next(() => this.referenceDelegate.removeReference(txn, key))
```

Unless you have data that the `waitFor` parallelism meaningfully improves performance :)

Reply:
There are only two promises here, and one of them is a delete for a single key.